Shared/Samples/Browse WMS layers/BrowseWMSLayersView.swift

```diff
@@ -0,0 +1,193 @@
+            }
+            .errorAlert(presentingError: $error)
+    }
+}
+
+extension BrowseWMSLayersView {
+    /// A view that displays the layers and sublayers in a hierarchical list.
+    struct WMSLayerListView: View {
+        /// The models to display in the list.
+        let models: [WMSLayerModel]
+        
+        /// The selected models that represent layer infos that we will display
+        /// in the `WMSLayer` on the map.
+        @Binding var selection: [WMSLayerModel]
+        
+        var body: some View {
+            List(models) { model in
+                OutlineGroup(models, children: \.children) { model in
+                    HStack {
+                        Text(model.layerInfo.title)
+                        Spacer()
+                        if model.kind == .display {
+                            Button {
+                                model.isVisible.toggle()
+                            } label: {
+                                Image(
+                                    systemName: model.isVisible || model.isParentVisible ? "eye" : "eye.slash"
+                                )
+                            }
+                            .buttonStyle(.borderless)
+                            // Disable the button if the parent is visible because
+                            // the sublayer will always display in that case.
+                            .disabled(model.isParentVisible)
+                            .padding(.trailing)
+                        }
+                    }
+                    .font(.subheadline)
+                    .animation(.default, value: model.isVisible)
+                    .onChange(of: model.isVisible) { updateSelection() }
+                }
+            }
+            .listStyle(.plain)
+        }
+        
+        /// Updates the selection for given `isVisible`.
+        private func updateSelection() {
+            func visibleItems(startingWith model: WMSLayerModel) -> [WMSLayerModel] {
+                // If the starting one is visible, return that only because
+                // child visibility doesn't matter when the parent is visible.
+                guard !model.isVisible else { return [model] }
+                guard let children = model.children else { return [] }
+                return children.flatMap { visibleItems(startingWith: $0) }
+            }
+            
+            selection = models.flatMap { visibleItems(startingWith: $0) }
+        }
+    }
+}
+
+extension BrowseWMSLayersView {
+    /// The model for a WMS layer info.
+    @Observable
+    final class WMSLayerModel: Equatable, Identifiable {
+        /// The layer info that this model wraps.
+        let layerInfo: WMSLayerInfo
+        
+        /// Creates a model with a given WMS layer info.
+        init(layerInfo: WMSLayerInfo) {
+            self.layerInfo = layerInfo
+            children = layerInfo.sublayerInfos.map(WMSLayerModel.init(layerInfo:))
```

Comment:
One thing I noticed is that the leaf nodes in the outline group have the disclosure arrow >. From the doc of [`OutlineGroup`](https://developer.apple.com/documentation/swiftui/outlinegroup), if the `children` property is `nil`, it will be shown as a leaf node (without >). So maybe something like this

```swift
            if !layerInfo.sublayerInfos.isEmpty {
                children = layerInfo.sublayerInfos.map(WMSLayerModel.init(layerInfo:))
            } else {
                children = nil
            }
```

Reply:
Great suggestion